src/python_pachyderm/mixin/util.py

```diff
@@ -1,18 +1,35 @@
+import re
+
 from python_pachyderm.proto.v2.pfs import pfs_pb2 as pfs_proto
 
+# copied from pachyderm/pachyderm
+uuid_re = re.compile(r"[0-9a-f]{12}4[0-9a-f]{19}")
+valid_branch_re = re.compile(r"^[a-zA-Z0-9_-]+$")
+
 
-def commit_from(src, allow_just_repo=False):
+def commit_from(src):
+    # TODO: refactor this overloading of repo/commit in a single string
     if src is None:
         return None
-
     if isinstance(src, pfs_proto.Commit):
         return src
-    elif isinstance(src, (tuple, list)) and len(src) == 2:
-        return pfs_proto.Commit(repo=pfs_proto.Repo(name=src[0]), id=src[1])
+
+    repo, branch, commit = None, None, None
+    if isinstance(src, (tuple, list)) and len(src) == 2:
+        repo, branch_or_commit = src
     elif isinstance(src, str):
-        repo_name, commit_id = src.split("/", 1)
-        return pfs_proto.Commit(repo=pfs_proto.Repo(name=repo_name), id=commit_id)
+        repo, branch_or_commit = src.split("/", 1)
+    if uuid_re.match(branch_or_commit) or not valid_branch_re.match(branch_or_commit):
+        commit = branch_or_commit
+    else:
+        branch = branch_or_commit
+
+    if repo is None:
+        raise ValueError("Repo cannot be empty")
 
-    if not allow_just_repo:
-        raise ValueError("Invalid commit type")
-    return pfs_proto.Commit(repo=pfs_proto.Repo(name=src))
+    return pfs_proto.Commit(
+        id=commit,
```

Comment:
Does this parse `src` differently than the old version, or is it just the output that changed?

I think it might make sense rid of this and focus on moving to the `(repo, branch, commit)` syntax in the calls, since I think there's no way to use this to see commits other than the most recent

Reply:
Only the output has changed. I can also drop support for `"repo/commit-or-branch"` syntax here.